From 4880fa3ec6c0a4cb070b45f42abb63424f9bd593 Mon Sep 17 00:00:00 2001
From: Everett Hildenbrandt <hildenb2@illinois.edu>
Date: Wed, 27 Sep 2017 15:44:48 -0500
Subject: [PATCH] build: building cvc4 before building Maude

CVC4 is not available in every system's package manager, so we build it directly from the already cloned git submodule.
Note that CVC4 needs Antlr 3.4, so we use the supplied script for downloading and building that as well.
---
 build | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/build b/build
index cfeba0ac..ad5f9e19 100755
--- a/build
+++ b/build
@@ -8,6 +8,16 @@ build_dir="$top_dir/.build"
 include_dir="$build_dir/include"
 lib_dir="$build_dir/lib"
 
+git submodule update --init
+
+( cd src/SMT/cvc4 \
+    && ./autogen.sh \
+    && ./contrib/get-antlr-3.4 \
+    && ./configure --prefix=$build_dir \
+                   --with-antlr-dir=$(pwd)/antlr-3.4 ANTLR=$(pwd)/antlr-3.4/bin/antlr3 \
+    && make -s -j4 \
+    && make -s install )
+
 automake --add-missing
 
 aclocal \
-- 
2.13.5

